core/src/toga/app.py

```diff
@@ -244,35 +244,43 @@ def __init__(
         description: str | None = None,
         startup: AppStartupMethod | None = None,
         on_exit: OnExitHandler | None = None,
-        windows: Iterable[Window] = (),  # DEPRECATED
```

Comment:
This PR is moved `windows` to deprecated; I agree it should be a no-op, but I don't think we can delete it outright.

Reply:
As discussed, it shouldn't be a no-op because that would silently break code that used to work. Instead, I'll make it throw an exception with a better message than the default one.